src/amo/components/Collection/index.js

```diff
@@ -121,9 +111,15 @@ export class CollectionBase extends React.Component<Props> {
       collectionChanged = true;
     }
 
+    // When switching into edit mode, refresh the collection add-ons and
+    // show page 1.
+    if (this.props.editing && !nextProps) {
+      addonsPageChanged = true;
+      location.query.page = 1;
```

Comment:
I don't fully understand this, but I've made the change in the way (I think) you suggested and it works, so you can let me know if it's what you were thinking.

Reply:
We cannot really change `location` like this because I don't think a redux action will be dispatched (and the routing layer is hooked into redux). We should the router to update an URL.